manifests/settings_file.pp

```diff
@@ -12,13 +13,31 @@
 # mode: settings file's mode
 #
 define foreman_proxy::settings_file (
-    $path = "/etc/foreman-proxy/settings.d/${title}.yml",
+    $enabled       = undef,
+    $listen_on     = undef,
+    $path          = "/etc/foreman-proxy/settings.d/${title}.yml",
+    $owner         = 'root',
+    $group         = $::foreman_proxy::user,
+    $mode          = '0640',
     $template_path = "foreman_proxy/${title}.yml.erb",
-    $owner = 'root',
-    $group = $::foreman_proxy::user,
-    $mode = '0640',
   ) {
 
+  if $title != 'settings' {
+    $real_enabled = pick($enabled, getvar("foreman_proxy::${title}"))
+    $real_listen_on = pick($listen_on, getvar("foreman_proxy::${title}_listen_on"))
+
```

Comment:
I'm not sure about the if and wonder if we should just make the main settings.yml explicit instead of this settings file. An alternative would be to create a plugin_settings_file define. Any ideas what would be cleaner?